docs/basic-features/data-fetching/incremental-static-regeneration.md

```diff
@@ -83,6 +83,40 @@ When a request is made to a page that was pre-rendered at build time, it will in
 - Next.js triggers a regeneration of the page in the background.
 - Once the page has been successfully generated, Next.js will invalidate the cache and show the updated page. If the background regeneration fails, the old page would still be unaltered.
 
-When a request is made to a path that hasn’t been generated, Next.js will server-render the page on the first request. Future requests will serve the static file from the cache.
+When a request is made to a path that hasn’t been generated, Next.js will server-render the page on the first request. Future requests will serve the static file from the cache. ISR on Vercel [persists the cache globally and handles rollbacks](https://vercel.com/docs/concepts/next.js/incremental-static-regeneration).
 
-[Incremental Static Regeneration](https://vercel.com/docs/concepts/next.js/incremental-static-regeneration) covers how to persist the cache globally and handle rollbacks.
+## Error Handling and Revalidation
+
+If there is an error inside `getStaticProps`/`getStaticPaths` when handling background regeneration, or you manually throw an error, the last successfully generated page will continue to be shown. On the next subsequent request, Next.js will retry calling `getStaticProps`/`getStaticPaths`.
+
+```jsx
+export async function getStaticProps() {
+  // If this request throws an uncaught error, Next.js will
+  // not invalidate the currently shown page and
+  // retry getStaticProps on the next request.
+  const res = await fetch('https://.../posts')
+  const posts = await res.json()
+
+  if (res.status >= 500) {
```

Comment:
Would we have gotten back `posts` which is returned below if a `500` status code was returned? 

I think it might be good to discourage returning/changing revalidate in the error case.

Reply:
@ijjk, on `returning/changing`, can you elaborate more on what the value for `relvalidate` should be or shall we remove it completely?

Reply:
Thanks @ijjk, I'll update the code snippet.